build/push-docker-deploy.sh

```diff
@@ -12,11 +12,10 @@ run/local-cluster.sh stop
 docker tag cockroachdb/cockroach:latest cockroachdb/cockroach:${VERSION}
 docker tag cockroachdb/cockroach-dev:latest cockroachdb/cockroach-dev:${VERSION}
 
-# Pushing to the registry just fails sometimes, so for the time
-# being just make this a best-effort action.
-set +e
-
-docker push cockroachdb/cockroach:latest
-docker push cockroachdb/cockroach:${VERSION}
-docker push cockroachdb/cockroach-dev:latest
-docker push cockroachdb/cockroach-dev:${VERSION}
+for type in {,-dev}; do
+  for version in {latest, ${VERSION}}; do
+    # Pushing to the registry just fails sometimes, so for the time
+    # being just make this a best-effort action.
+    docker push cockroachdb/cockroach${type}:${version} || true
+  done
+done
```

Comment:
why not ignore this in `circleci.yml`? seems weird that a script push-docker-deploy.sh doesn't actually make sure the push works.